src/main/java/com/github/onsdigital/dp/uploadservice/api/APIClient.java

```
@@ -1,4 +1,132 @@
+            builder.setBoundary("--TFJ5T8Nl2Py-S_BZXD5_FaEzCCuRXVXL0--[\\r][\\n]");
+            final HttpEntity entityReq = builder.build();
+            request.setEntity(entityReq);
+            request.addHeader("Authorization", "Bearer " + authToken);
+            httpClient.execute(request);
+        } catch (Exception e) {
+            throw new ConnectionException("error talking to upload service", e);
+        }
+    }
+
+    @Override
+    public void uploadResumableFile(File file, List<NameValuePair> params) {
+        long fileSize = file.length();
+        int totalChunks = (int) Math.ceil((double) fileSize / CHUNK_SIZE);
+
+        try (FileInputStream fileInputStream = new FileInputStream(file)) {
+
+            params.add(new BasicNameValuePair("resumableTotalSize", String.valueOf(fileSize))); // the total size of the file
+            params.add(new BasicNameValuePair("resumableTotalChunks", String.valueOf(totalChunks)));
+
+            for (int chunkNumber = 0; chunkNumber < totalChunks; chunkNumber++) {
+                int start = chunkNumber * CHUNK_SIZE;
+                int resumableChunkSize = Math.min(CHUNK_SIZE, (int) (fileSize - start));
+                byte[] buffer = new byte[resumableChunkSize];
+
+                int bytesRead = fileInputStream.read(buffer, 0, resumableChunkSize);
+                if (bytesRead == -1) break;
+
+                params.add(new BasicNameValuePair("resumableChunkNumber", String.valueOf(chunkNumber+1)));
+                params.add(new BasicNameValuePair("resumableChunkSize", String.valueOf(resumableChunkSize)));
+
+                sendChunk(buffer, chunkNumber, totalChunks, file.getName(), params);
+
+                params.removeIf(nameValuePair -> nameValuePair.getName().equals("resumableChunkNumber"));
+                params.removeIf(nameValuePair -> nameValuePair.getName().equals("resumableChunkSize"));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

This should be thrown as an IO exception rather than just printing out the stack trace

---

This is now addressed.